src/huggingface_hub/hf_api.py

```diff
@@ -3542,6 +3542,34 @@ def update_repo_visibility(
         hf_raise_for_status(r)
         return r.json()
 
+    @validate_hf_hub_args
+    def update_repo_settings(
+        self,
+        repo_id: str,
+        gated: Literal["auto", "manual", False] = False,
+        *,
```

Comment:
```suggestion
        *,
        gated: Literal["auto", "manual", False] = False,
```

Reply:
Okay, only these two modifications are required?

Reply:
I think so but I'll have to re-review more in depth once applied (I did not try it locally yet)

Reply:
Okay, I'll commit these changes. After you've tested them locally, feel free to let me know if any further adjustments are needed.